### PR TITLE
Render Slack Attachments

### PR DESCRIPTION
**Pull request by kevsmith**
This PR adds attachment rendering support to `Cog.Chat.Slack.TemplateProcessor` and tweaks to `Slack.Connector`, `Slack.Provider`, and a fork of `Elixir-Slack` to send the rendered attachments to Slack.

Also added a utility module, `Cog.Util.Colors`, which translates CSS color names to their corresponding hex codes. This should make working with colors more natural for command authors.

Hopefully the `Elixir-Slack` fork is short lived and my small change is upstreamed.

### CODE REVIEW

File: lib/cog/chat/slack/template_processor.ex [hunk 1]
```diff
@@ -1,8 +1,16 @@
...
-  def render(directives),
-    do: Enum.map_join(directives, &process_directive/1)
+  alias Cog.Util.Colors
+
+  def render(directives) do
+    {text, attachments} = directives
+                          |> Enum.map(&process_directive/1)
+                          |> List.flatten
+                          |> consolidate_outputs({"", []})
+  {text, Enum.map(attachments, fn(attachment) -> updated = Map.delete(attachment, "name")
```
Comment (christophermaier) [inline lib/cog/chat/slack/template_processor.ex]: Perhaps some comments describing why we do this bit of munging, with pointers to the relevant Slack API docs?

File: lib/cog/util/colors.ex [hunk 2]
```diff
@@ -1,5 +1,8 @@
 defmodule Cog.Util.Colors do
-  hex_codes = File.read!("priv/css-color-names.json") |> Poison.decode!
+
+  @external_resource "priv/css-color-names.json"
```
Comment (christophermaier) [inline lib/cog/util/colors.ex]: Should we use `:code.priv_dir(:cog)` here instead?

**Comment by imbriaco** [comment]
Looking good!

I think we want to set the `fallback` attribute to the same content as the `text` attribute when we send it to Slack's API. That content is displayed on devices that aren't the Slack web client like mobile and IRC.

**Comment by christophermaier** [comment]
👍

**Pull request merged**